src/com/gfredericks/schpec/number.clj

```diff
@@ -0,0 +1,119 @@
+(ns com.gfredericks.schpec.number
+  (:require [clojure.spec :as s]
+            [clojure.spec.gen :as gen])
+  (:import [java.math BigDecimal MathContext RoundingMode]))
+
+(defn real?
+  [x]
+  (and (number? x)
+       (case x
+         Double/POSITIVE_INFINITY false
+         Double/NEGATIVE_INFINITY false
+         Double/NaN false
+         true)))
+
+(s/def ::real
+  (s/spec real?
+          :gen #(gen/double* {:infinite? false :NaN? false})))
```

Comment:
Is this spec/pred intended to apply just to doubles? If so, I would use a name closer to "double" or "floating-point", and check that it's an instance of `Double` in the predicate; I personally don't like the name "real" for numeric types because of the mismatch with the mathematical term.